Gp 21 fe landing page

thuanvmse161224 opened the pull request.

thuanvmse161224 committed:
feat: adding Landing Page
1. Add the entry into PublicRouter
2. Adding basic Landing Page layout

thuanvmse161224 committed:
Merge branch 'master' of https://github.com/FPTU-GHub-Capstone/ghub-app into GP-21-fe-landing-page

thuanvmse161224 committed:
feat: add contact form
- Contact form in its early stage
- Didnt have hookform validation
- Still hasnt match with figma

thuanvmse161224 committed:
refactor: reorganize textfields
- textfield are being organize to its respective components
- FirstName, LastName, Email, Message
- Less clutter in contactForm
- minor change in index.html logo show

thuanvmse161224 committed:
fix: fix conflict, and merge new master into

thuanvmse161224 committed:
fix: fix routse string

danielNg91 commented on src/pages/Landing/Contact/ContactForm/components/EmailTextField.tsx:
```diff
@@ -0,0 +1,36 @@
+};
+
+// eslint-disable-next-line no-useless-escape
+const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
```
this regex can be reused, put it in common or contants.ts and import it

thuanvmse161224 replied:
Done

danielNg91 commented on src/pages/Landing/Landing.tsx:
```diff
@@ -0,0 +1,100 @@
+export const Landing: React.FC = () => {
+	return (
+		<>
+			<div style={{ backgroundColor: '#39289F', padding: '5vw'}} id='landing-main'>
```
with lots of styles here, it's  better to be create a styled component for this div

thuanvmse161224 replied:
done

thuanvmse161224 committed:
chore: remove unused imports

thuanvmse161224 committed:
refactor: extract regex patterns into common/regex

thuanvmse161224 committed:
refactor: cleaning and extract various components
- Extract the whychoose-us section
- Extract type, data, component of whychooseus, for greater visibility
- Extract styles of Landing
